lib/scala/kg/src/main/scala/io/github/tetherlessworld/mcsapps/lib/kg/stores/neo4j/Neo4jKgQueryStore.scala

```diff
@@ -293,19 +293,39 @@ final class Neo4jKgQueryStore @Inject()(configuration: Neo4jStoreConfiguration)
       val cypher = KgNodeQueryFulltextCypher(query)
 
       // Get sources
-      val sourceIds =
+      val sourceIdStringFacets =
         transaction.run(
           s"""
              |${cypher}
              |MATCH (node)-[:${SourceRelationshipType}]->(source:${SourceLabel})
-             |RETURN DISTINCT source.id
+             |RETURN DISTINCT source.id, COUNT(*) AS count
              |""".stripMargin,
           toTransactionRunParameters(cypher.bindings)
-        ).asScala.map(record => record.get("source.id").asString()).toSet.toList
+        ).asScala.map(record => StringFacet(
+          count = record.get("count").asInt,
+          value = record.get("source.id").asString
+        )).toList
+
+      val searchResultTypeFacets =
+        transaction.run(
+          s"""
+             |${cypher}
+             |RETURN labels(node)[0] AS type, COUNT(*) AS count
+             |""".stripMargin,
+          toTransactionRunParameters(cypher.bindings)
+        ).asScala.map(record => KgSearchResultTypeFacet(
+          count = record.get("count").asInt,
+          value = record.get("type").asString match {
```

Comment:
Use KgSearchResultType.values.find(...).getOrElse instead of hard-coding the enumerators here.

Reply:
Changed